### PR TITLE
chore: Update fork with latest fixes from source project

### PR DESCRIPTION
**Pull request by mauriciosierra**
Fixes https://github.com/Shopify/react-native-packages/issues/4074

### Motivation and Context
There are a couple of deprecated gradle properties that have been updated in the source project. This PR just brings in those updates to our fork version of this lib.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -8,7 +8,7 @@ buildscript {
         google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.2.1'
+        classpath 'com.android.tools.build:gradle:8.0.1'
```
Comment (tapi) [inline build.gradle:11]: Does this require us to already be on Gradle 8?
https://developer.android.com/build/releases/gradle-plugin#updating-gradle

Comment (mauriciosierra) [reply build.gradle:11]: Hmm.. maybe. Let me see if I can somehow test from a POS build to use this version. The alternative could be to make a temporary commit in our fork to update only the deprecated properties so we can finish all the other work, bump our gradle version, and then remove our temp commit and bring in the source updates into this fork?

Comment (tapi) [reply build.gradle:11]: Thats an option, another possibility is to resolve all other deps in POS then include this last.
Potential issue there is that we may also require an android gradle plugin update as well which makes the change in POS a lot less isolated. I know you guys are gonna need it eventually for the Stripe SDK but my preference from learned from the 1.6 update is to keep changes separate

Comment (mauriciosierra) [reply build.gradle:11]: I ran a test by pointing our dependency of this library to the main project at the latest version (`7.1.3`) and this built and ran fine, and I was able to authenticate successfully. So, I think that applying this change would be fine to do before the Gradle 8 update.

**Review by michaelrhughes — APPROVED** [review]
Changes LGTM, it would be good to also bump our fork of `react-native-app-auth` before actually shipping this

**Comment by mauriciosierra** [comment]
This PR doesn't need to be merged as per @michaelrhughes

**Pull request closed without merging**